src/socket.io/index.js

```diff
@@ -196,18 +197,23 @@ function authorize(socket, callback) {
 			cookieParser(request, {}, next);
 		},
 		function (next) {
-			db.sessionStore.get(request.signedCookies[nconf.get('sessionKey')], function (err, sessionData) {
-				if (err) {
-					return next(err);
-				}
-				if (sessionData && sessionData.passport && sessionData.passport.user) {
-					request.session = sessionData;
-					socket.uid = parseInt(sessionData.passport.user, 10);
+			 if(socket && socket.request && socket.request.cookies && socket.request.cookies.token){
+					 var username = jwt.verify(socket.request.cookies.token, 'secret').username || '';
```

Comment:
The value `"secret"` should not be hardcoded. Please try to import this from the `nconf` module. Also, fix the same vulnerability on Open edX side.